### PR TITLE
introduce identity function

### PR DESCRIPTION
**Pull request by Tuxified**
as proposed by Alexey Nikitin (@tank-bohr ?), introduce the identity function as an alternative
to using `&(&1)`. 

edit:
if Alexey wants to create this PR, I'd be happy to delete mine

### CODE REVIEW

**Comment by tank-bohr** [comment]
@Tuxified Thank you very much. I appreciate it a lot. I think this PR is good enough :)

File: lib/elixir/lib/function.ex [hunk 1]
```diff
@@ -166,4 +166,27 @@ defmodule Function do
...
+  @doc """
+  Returns its input `value`, which can be given as an anonymous function
+  to transformation functions.
+
+  ## Examples
+
+      iex> Function.identity("Hello world!")
+      "Hello world!"
+
+      iex> 'abcdaabccc' |> Enum.sort |> Enum.chunk_by(&Function.identity/1)
```
Comment (thiamsantos) [inline lib/elixir/lib/function.ex]:
```suggestion
      iex> 'abcdaabccc' |> Enum.sort() |> Enum.chunk_by(&Function.identity/1)
```
:nail_care: I think we could use in the example a code following the default rules from the formatter.

File: lib/elixir/lib/function.ex [hunk 2]
```diff
@@ -166,4 +166,27 @@ defmodule Function do
...
+      ['aaa', 'bb', 'cccc', 'd']
+
+      iex> Enum.group_by('abracadabra', &Function.identity/1)
+      %{97 => 'aaaaa', 98 => 'bb', 99 => 'c', 100 => 'd', 114 => 'rr'}
+
+      iex> Enum.map([1, 2, 3, 4], &Function.identity/1)
+      [1, 2, 3, 4]
+
+  """
+  @doc since: "1.10.0"
```
Comment (eksperimental) [inline lib/elixir/lib/function.ex]: Shouldn't it just be "1.10"?

Comment (josevalim) [reply lib/elixir/lib/function.ex]: No, it is always fully specified, because something could be added on 1.10.2, for example. It is rare though.

**Comment by OvermindDL1** [comment]
There could be a `Function.identity/0` that just returns `&Function.identity/1` as well.  If this is going to become a theme, what about other generic and standard functions like `Function.const/1,2` and so forth, the general full usual 'set' of them?

File: lib/elixir/lib/function.ex (hunk 2 above)
Comment (eksperimental) [reply lib/elixir/lib/function.ex]: I agree, but the all references to x.0 should be replaced with x.0.0

Comment (eksperimental) [reply lib/elixir/lib/function.ex]: Unless that's not the way we are doing it

**Comment by josevalim** [comment]
@OvermindDL1 please submit a proposal to the mailing list but it is unlikely that we will be adding functions that return functions.

File: lib/elixir/lib/function.ex (hunk 2 above)
Comment (josevalim) [reply lib/elixir/lib/function.ex]: I can't think of anywhere we used `since: "x.y"`.

Comment (eksperimental) [reply lib/elixir/lib/function.ex]: Yes, you are right. I think i got confused the the two zeroes in 10.0

Comment (eksperimental) [reply lib/elixir/lib/function.ex]: Yes, you are right. I think i got confused the the two zeroes in 10.0

**Comment by ericmj** [comment]
Thank you @Tuxified!

**Pull request merged**